scripts/check-queries.lua

```diff
@@ -1,13 +1,48 @@
 -- Execute as `nvim --headless -c "luafile ./scripts/check-queries.lua"`
+
+local function extract_captures()
+  local lines = vim.fn.readfile("CONTRIBUTING.md")
+  local captures = {}
+  local current_query
+
+  for _, line in ipairs(lines) do
+    if vim.startswith(line, "### ") then
+      current_query = vim.fn.tolower(line:sub(5))
+    elseif vim.startswith(line, "@") and current_query then
+      if not captures[current_query] then
+        captures[current_query] = {}
+      end
+
+      table.insert(captures[current_query], vim.split(line:sub(2), " ", true)[1])
+    end
+  end
+
+  return captures
+end
+
 local function do_check()
   local parsers = require 'nvim-treesitter.parsers'.available_parsers()
   local queries = require 'nvim-treesitter.query'
   local query_types = queries.built_in_query_groups
 
+  local captures = extract_captures()
```

Comment:
shouldn't we check directly from `TSHighlighter.hl_map` ?

Reply:
No because we also want to check the `locals` and `fold` queries, and they are not contained in the `hl_map`. Maybe we should have an easy way for modules to declare the captures they use.